lexers/embedded/hare.xml

```diff
@@ -0,0 +1,101 @@
+<lexer>
+  <config>
+    <name>Hare</name>
+    <alias>hare</alias>
+    <filename>*.ha</filename>
+    <mime_type>text/x-hare</mime_type>
+  </config>
+  <rules>
+    <state name="whitespace">
+      <rule pattern="[\s\n]+">
+        <token type="TextWhitespace"/>
+      </rule>
+      <rule pattern="@[a-z]+">
+        <token type="NameDecorator"/>
+      </rule>
+      <rule pattern="//.*\n">
+        <token type="CommentSingle"/>
+      </rule>
+    </state>
+    <state name="statements">
+      <rule pattern="&quot;">
+        <token type="LiteralString"/>
+        <push state="string"/>
+      </rule>
+      <rule pattern="`[^`]*`">
+        <token type="LiteralString"/>
+      </rule>
+      <rule pattern="'(\\[\\0abfnrtv&quot;']||\\(x[a-fA-F0-9]{2}|u[a-fA-F0-9]{4}|U[a-fA-F0-9]{8})|[^\\'])'">
+        <token type="LiteralStringChar"/>
+      </rule>
+      <rule pattern="(0|[1-9]\d*)\.\d+([eE][+-]?\d+)?(f32|f64)?">
+        <token type="LiteralNumberFloat"/>
+      </rule>
+      <rule pattern="(0|[1-9]\d*)([eE][+-]?\d+)?(f32|f64)">
+        <token type="LiteralNumberFloat"/>
+      </rule>
+      <rule pattern="0x[0-9a-fA-F]+\.[0-9a-fA-F]+([pP][+-]?\d+(f32|f64)?)?">
+        <token type="LiteralNumberFloat"/>
+      </rule>
+      <rule pattern="0x[0-9a-fA-F]+[pP][+-]?\d+(f32|f64)">
+        <token type="LiteralNumberFloat"/>
+      </rule>
+      <rule pattern="0x[0-9a-fA-F]+(z|[iu](8|16|32|64)?)?">
+        <token type="LiteralNumberHex"/>
+      </rule>
+      <rule pattern="0o[0-7]+(z|[iu](8|16|32|64)?)?">
+        <token type="LiteralNumberOct"/>
+      </rule>
+      <rule pattern="0b[01]+(z[iu](8|16|32|64)?)?">
```

Comment:
This doesn't highlight `0b1010u16` properly, as well as incorrectly catching `0b1010zu8`.

```suggestion
      <rule pattern="0b[01]+(z|[iu](8|16|32|64)?)?">
```